### PR TITLE
Remove BC Layer for Symfony Translator versions 3 and lower

### PR DESCRIPTION
**Pull request by rpkamp**

### CODE REVIEW

**Comment by rpkamp** [comment]
This PR partially resolves https://github.com/Behat/Behat/issues/1305

File: src/Behat/Behat/Definition/Translator/Translator.php [hunk 1]
```diff
@@ -4,12 +4,4 @@
...
 {
-    
-    public function trans($id, array $parameters = array(), $domain = null, $locale = null ) {
-        
-        if(array_key_exists('%count%', $parameters) && method_exists($this, 'transChoice') && !$this instanceof \Symfony\Contracts\Translation\TranslatorInterface){
-            return parent::transChoice($id, $parameters['%count%'], $parameters, $domain, $locale);
-        }
-        return parent::trans($id, $parameters, $domain, $locale);
-    }
 }
```
Comment (Taluu) [inline src/Behat/Behat/Definition/Translator/Translator.php:7]: you should probably add a deprecation for this class then

Comment (rpkamp) [reply src/Behat/Behat/Definition/Translator/Translator.php:7]: It's still interesting to keep it as an internal stable interface I think.

See https://github.com/Behat/Behat/issues/1316

**Comment by rpkamp** [comment]
Anything needed from me here to push this forward?

**Comment by ciaranmcnulty** [comment]
Thanks!

**Pull request merged**